src/naip_cnn/data.py

```diff
@@ -116,46 +130,121 @@ def _load(self, shuffle: bool = True, seed: int = 0) -> tf.data.Dataset:
 
 
-class NAIPDataset(HDF5Dataset):
-    """A dataset of NAIP images and associated forest attribute labels."""
+class TrainingDataset(HDF5Dataset):
+    """A dataset of NAIP images and associated forest attribute labels.
+
+    This class is used both to define and generate sample datasets and to load
+    previously generated datasets.
+    """
 
     def __init__(
         self,
-        shape: tuple[int, int] = (150, 150),
-        name: str = "MAL2016_CanyonCreek",
-        label: str = "cover",
-        root_dir: str = "../data/training",
+        acquisition: Acquisition,
+        naip_res=1.0,
+        lidar_res=30.0,
+        footprint: tuple[int, int] = (150, 150),
+        spacing: float = None,
         **kwargs,
     ) -> None:
         """
         Parameters
         ----------
-        shape : tuple[int, int]
-            The shape of the image, in pixels.
-        name : str
-            The name of the dataset to load.
-        label : str
-            The name of the label variable.
-        root_dir : str
-            The root directory of the data.
+        acquisition : Acquisition
+            The NAIP-LiDAR acquisition used to build the dataset.
+        naip_res : float
+            The NAIP resolution in meters.
+        lidar_res : float
+            The LiDAR resolution in meters.
+        footprint : tuple[int, int]
+            The size of the sampled footprints in meters.
+        spacing : float
+            The spacing between sampled footprints in meters. If None, defaults to
+            the footprint width.
         **kwargs
-            Additional arguments passed to HDF5Dataset.
+            Additional arguments passed to HDF5Dataset, e.g. train_split, val_split,
         """
-        self.shape = shape
-        self.name = name
-        self.root_dir = Path(root_dir)
-
-        str_shape = f"{self.shape[0]}x{self.shape[1]}"
-        path = self.root_dir / ("_".join([self.name, str_shape]) + ".h5")
-        super().__init__(
-            path=path.as_posix(), feature_name="image", label_name=label, **kwargs
+        self.acquisition = acquisition
+        self.naip_res = naip_res
+        self.lidar_res = lidar_res
+        self.footprint = footprint
+        self.spacing = spacing if spacing is not None else footprint[0]
+
+        super().__init__(path=self.hdf_path, feature_name="image", **kwargs)
+
+    def __repr__(self) -> str:
+        return f"<TrainingDataset name={self.name}>"
+
+    @property
+    def name(self):
+        footprint_str = f"{self.footprint[0]}x{self.footprint[1]}"
+        naip_res_str = float_to_str(self.naip_res)
+        lidar_res_str = float_to_str(self.lidar_res)
+        spacing_str = float_to_str(self.spacing)
+        return "-".join(
+            [
+                self.acquisition.name,
+                naip_res_str,
+                lidar_res_str,
+                footprint_str,
+                spacing_str,
+            ]
+        )

```

Comment:
Yep, as you say, kind of a beastly file naming convention, but I think all attributes are needed to differentiate between possible parameters.  The only other thought I had was to have a sidecar file (JSON or YAML) that would store some of these attributes, but that becomes difficult when you're using this as a download name to kick off a GEE export and presumably you'd have to write this sidecar file locally, so they would become disconnected.  I don't have too much to offer on this one.

Reply:
A sidecar JSON file could definitely work, but as you mentioned there's some complexity with how you name the exported data. I'm hoping this is as long as the file names will have to get for the training datasets, although for the `ModelRun` in particular I'm a little worried about how we'll handle runs with different hyperparameters. I'll keep the sidecar idea in mind and may have to resort to that if things get too unwieldy in the future.